hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -409,31 +398,29 @@ public HoodieMetadataPayload preCombine(HoodieMetadataPayload previousRecord) {
     checkArgument(previousRecord.key.equals(key),
         "Cannot combine " + previousRecord.key + " with " + key + " as the keys differ");
 
-    switch (type) {
-      case METADATA_TYPE_PARTITION_LIST:
-      case METADATA_TYPE_FILE_LIST:
-        Map<String, HoodieMetadataFileInfo> combinedFileInfo = combineFileSystemMetadata(previousRecord);
-        return new HoodieMetadataPayload(key, type, combinedFileInfo);
-      case METADATA_TYPE_BLOOM_FILTER:
-        HoodieMetadataBloomFilter combineBloomFilterMetadata = combineBloomFilterMetadata(previousRecord);
-        return new HoodieMetadataPayload(key, combineBloomFilterMetadata);
-      case METADATA_TYPE_COLUMN_STATS:
-        return new HoodieMetadataPayload(key, combineColumnStatsMetadata(previousRecord));
-      case METADATA_TYPE_RECORD_INDEX:
-        // There is always a single mapping and the latest mapping is maintained.
-        // Mappings in record index can change in two scenarios:
-        // 1. A key deleted from dataset and then added again (new filedID)
-        // 2. A key moved to a different file due to clustering
-
-        // No need to merge with previous record index, always pick the latest payload.
-        return this;
-      case METADATA_TYPE_SECONDARY_INDEX:
-        // Secondary Index combine()/merge() always returns the current (*this*)
-        // record and discards the prevRecord. Based on the 'isDeleted' marker in the payload,
-        // the merger running on top takes the right action (discard current or retain current record).
-        return this;
-      default:
-        throw new HoodieMetadataException("Unknown type of HoodieMetadataPayload: " + type);
+    if (type == MetadataPartitionType.FILES.getRecordType() || type == MetadataPartitionType.FILES.getRecordType(RECORDKEY_PARTITION_LIST)) {
+      Map<String, HoodieMetadataFileInfo> combinedFileInfo = combineFileSystemMetadata(previousRecord);
```

Comment:
Similar here on removing the if-else logic by replacing it with logic in each enum in `MetadataPartitionType`.